OpenRA.Game/Support/Log.cs

```diff
@@ -75,22 +77,33 @@ public static void AddChannel(string channelName, string baseFilename)
 			}
 		}
 
-		public static void Write(string channel, string value)
+		public static void Write(string channelName, string value)
 		{
-			var writer = Channel(channel).Writer;
+			var channel = Channels[channelName];
+			var writer = channel.Writer;
 			if (writer == null)
 				return;
 
-			writer.WriteLine(value);
+			if (!channel.IsTimestamped)
+				writer.WriteLine(value);
+			else
+				writer.WriteLine("[{0}] {1}", DateTime.Now.ToString(Game.Settings.Server.TimestampFormat), value);
 		}
 
-		public static void Write(string channel, string format, params object[] args)
+		public static void Write(string channelName, string format, params object[] args)
 		{
-			var writer = Channel(channel).Writer;
-			if (writer == null)
+			var channel = Channels[channelName];
```

Comment:
Please change this to `var channel =  Channel(channel)`. The `Channel()` method uses locking to avoid race conditions and gives a more readable error message when someone tries to log to an undefined channel.

Reply:
Done.